compiler/src/prog.ml

```diff
@@ -503,6 +503,12 @@ and spilled_c s c =  List.fold_left spilled_i s c
 
 let spilled fc = spilled_c Sv.empty fc.f_body
 
+let assigns = function
+  | Cassgn (x, _, _, _) -> written_lv Sv.empty x
+  | Copn (xs, _, _, _) | Csyscall (xs, _, _) | Ccall (xs, _, _) ->
+      List.fold_left written_lv Sv.empty xs
+  | Cif _ | Cfor _ | Cwhile _ -> Sv.empty
```

Comment:
where the assign of the for loop variable is taken into account ?

Reply:
The assignments to the for loop variable is not taken into account. Should this be changed? documented? raise an error?

The only use of this function is in register allocation, after for loops have been removed, so it does not matter much.

Reply:
Ok. Maybe in that case, an assert false for the case of Cfor is better. It will prevent bad usage in the future.